api/src/main/java/org/datacleaner/configuration/RemoteServerData.java

```diff
@@ -33,4 +33,8 @@
     public String getUsername();
 
     public String getPassword();
+
+    public void setUsername(String username);
```

Comment:
Maybe we can reach a compromise by removing the setters from the interface at least. Then you need to do an ```instanceof RemoteServerDataImpl``` check and in case it returns false throw an UnsupportedOperationException or so. But that makes the mutability "hidden" for the moment which is good for the future if we want to enforce immutability.